### ISSUE TITLE
Remove comments in extern

### ISSUE THREAD

**Issue by giraffate** [rust-lang/rustfmt#4528]
**Describe the bug**

rustfmt removes comments in extern. This seems to be introduced since v1.4.26. I found this in clippy CI https://github.com/rust-lang/rust-clippy/pull/6336/checks?check_run_id=1404315427 in PR https://github.com/rust-lang/rust-clippy/pull/6336.

**To Reproduce**

```
$ cd rust-clippy
$ path/to/rustfmt/target/debug/rustfmt/target/debug/rustfmt tests/ui/cast_ref_to_mut.rs --check
Diff in /Users/tnakata/workspace/rust-clippy/tests/ui/cast_ref_to_mut.rs at line 2:
 #![allow(clippy::no_effect)]
 
 extern "C" {
-    // N.B., mutability can be easily incorrect in FFI calls -- as
-    // in C, the default is mutable pointers.
     fn ffi(c: *mut u8);
     fn int_ffi(c: *mut i32);
 }

```

**Expected behavior**

not to remove comments

**Meta**

- rustfmt version: 
```
$ target/debug/rustfmt -V                                
rustfmt 1.4.26-nightly (293d7d01 2020-11-14)
```
- From where did you install rustfmt?: 
```
$ cd rustfmt
$ git checkout refs/tags/v1.4.26
$ CFG_RELEASE=1.45.0-nightly CFG_RELEASE_CHANNEL=nightly cargo build
$ target/debug/rustfmt -V                                
rustfmt 1.4.26-nightly (293d7d01 2020-11-14)
```
- How do you run rustfmt:
```
$ cd rust-clippy
$ path/to/rustfmt/target/debug/rustfmt/target/debug/rustfmt tests/ui/cast_ref_to_mut.rs --check
Diff in /Users/tnakata/workspace/rust-clippy/tests/ui/cast_ref_to_mut.rs at line 2:
 #![allow(clippy::no_effect)]
 
 extern "C" {
-    // N.B., mutability can be easily incorrect in FFI calls -- as
-    // in C, the default is mutable pointers.
     fn ffi(c: *mut u8);
     fn int_ffi(c: *mut i32);
 }
```

**Comment by calebcartwright** [comment]
Thanks for the report @giraffate, see the problem and will try to get the fix out to nightly ASAP

**Comment by faern** [comment]
It only remove multi line comments. Single line comments are preserved.

**Comment by berkus** [comment]
> It only remove multi line comments. Single line comments are preserved.

Not true.
![rustfmt_crazy](https://user-images.githubusercontent.com/80306/99446240-6d59d800-2927-11eb-97f2-9eae82f5cb7e.png)

**Comment by calebcartwright** [comment]
Closing as this is resolved in v1.4.27 available in the latest nightly (2020-11-18)
